lib/pf/ipset.pm

```diff
@@ -234,21 +239,53 @@ sub generate_mangle_postrouting_rules {
 
     my $rules = '';
 
+    my $indice = 100;
+    my $index = {};
+    my $indice2 = 1;
+
+    my @ints = split(',', $self->get_network_snat_interface());
+    push @ints,  $management_network->tag("int");
+    foreach my $int (@ints) {
+        $index->{$int} = $indice;
+        $indice --;
+    }
+
+    foreach my $interface (@internal_nets) {
+        my $dev = $interface->tag("int");
+        my $enforcement_type = $Config{"interface $dev"}{'enforcement'};
+        if (is_type_inline($enforcement_type)) {
+            $index->{$dev} = $indice2;
+            $indice2 ++;
+        }
+    }
+
     my @roles = pf::nodecategory::nodecategory_view_all;
 
-    foreach my $network ( keys %ConfigNetworks ) {
+    foreach my $network ( keys %NetworkConfig ) {
+
         next if ( !pf::config::is_network_type_inline($network) );
+        my $dev = $NetworkConfig{$network}{'interface'}{'int'};
+
+        my $gateway = (defined $NetworkConfig{$network}{'next_hop'} ? $NetworkConfig{$network}{'next_hop'} : $NetworkConfig{$network}{'gateway'});
+
+        my @interface_src = split(" ", pf_run("sudo ip route get 8.8.8.8 from $gateway"));
```

Comment:
Any specific reason for 8.8.8.8 ?

Probably would be better as a constant or even a configuration in the event the network cannot be routed to the internet

Reply:
We just try to find an network interface, 8.8.8.8 doesnt' really matter since we just need a public ip.